### PR TITLE
use ID defined in yaml instead of slug

### PR DESCRIPTION
**Pull request by dthiex**
Closes https://git.sinergise.com/team-6/openeo-platform/-/issues/179

### CODE REVIEW

**Comment by dthiex** [comment]
@janjezersek Do you think this is going to work? I am not sure about the part in the stacIndex.hbs. Obviously we would need to updated all yaml files before merging this.

Also do you think `OpenEOPName:` is a good name? Thinking about it, `OpenEOPID:` would maybe be better. I know we originally discussed to not introduce a new value and change the current one but the current one is based on the file name and I believe changing that would cause a lot of problems.

**Comment by dthiex** [comment]
I see what you mean and changes things accordingly. Does that make sense to you know or do you think it should be done differently. Our openEOP collections would then end up at .../testing/collections/SENTINEL2_L2A.

> I don't know how VITO is using it,

I found where they are using it but see if the id is something they need. I would suspect that the use this to map our collections to the correct openEO Platform collection name.

https://github.com/Open-EO/openeo-geopyspark-driver/blob/ad70925b6ccee0b7857b241b3c3a921246f0f158/openeogeotrellis/layercatalog.py#L674

Can you please check if you maybe understand more. Otherwise I will ask them after we agree internally on how to best approach this.

**Comment by Ardweaden** [comment]
> Does that make sense to you know or do you think it should be done differently

Looks fine to me.

> I found where they are using it but see if the id is something they need. I would suspect that the use this to map our collections to the correct openEO Platform collection name.

Yes, looks to be fine, they are basically doing the same as we do.

File: collections/e12c_motorway.yaml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 Name: E12C_Motorway
+OpenEOPID: E12C_Motorway
```
Comment (dthiex) [inline collections/e12c_motorway.yaml:2]: My mistake, names should always be all caps so it should be `E12C_MOTORWAY`. Same goes for `E12D_PRIMARY`.

**Review by dthiex — COMMENTED** [review]
@horvatdino I can't request changes because I am the author. Still, can you please change this.

File: collections/e12c_motorway.yaml (hunk 1 above)
Comment (horvatdino) [reply collections/e12c_motorway.yaml]: Done!

Comment (dthiex) [reply collections/e12c_motorway.yaml]: Thanks!

**Comment by Ardweaden** [comment]
> @Ardweaden Can you maybe double check if this looks good. I build it locally and also checked in the EDC code how we request the collection jsons and think it's fine but would be happy for another look of you. The new collection names are checked by me and are all correct.

Looks good to me, also tested it in a local deploy of our driver.

**Comment by dthiex** [comment]
@Ardweaden @horvatdino Can one of you approve this, so I can merge this.

**Review by horvatdino — APPROVED** [review]

**Review by Ardweaden — APPROVED** [review]

**Pull request merged**